src/server/ua_session.h

```diff
@@ -57,6 +57,9 @@ typedef struct {
     size_t localeIdsSize;
     UA_String *localeIds;
 
+    /* Role information */
+    UA_NodeId role;
```

Comment:
Does a session have only a single role?
I would guess that for fine-grained roles we would like to have several roles per session.

Reply:
I've incorporated managing multiple roles per session.

Reply:
Nice.
Since every role gets a power of two 2,4,8,16,32,… we can use a 64Bit integer as bitmap for all the roles. That is more efficient also for comparing inside the nodes..